code_challenges.rb

```diff
@@ -7,22 +7,78 @@
 fullArr2 = ['hello', 7, 23, -823, false, 78, nil, '67', 6, 'Number']
 # Expected output: [-823, 7, 23]
 
-
-
+# declare method only_odd_low_to_high that takes in one argument array
+def only_odd_low_to_high array
+    # select only values that are of class integer
+    only_ints = array.select do |num|
+        num.class == Integer
+    # now select only odd numbers
+    end.select do |num|
+        num.odd?
+    # use sort method to sort from low to high
+    end.sort
+    # return only the ints that are odd in ascending order
+    only_ints
+end 
+
+# p only_odd_low_to_high fullArr1
+# p only_odd_low_to_high fullArr2
 
 
 # --------------------2) Create a class called Bike that is initialized with a model, wheels, and a frame size. The default number of wheels is 2. Create a get_info method that returns a sentence with all the data from the bike object.
 
-
-
-
+# create class called Bike
+class Bike
+    # define the initialize with model, wheels, and frame size where default of wheels is 2
+    def initialize (model, frame_size)
+        @model = model
+        @wheels = 2
+        @frame_size = frame_size
+        # (3) add a bell and don't sent to anything
+        @bell = nil
+        # (4) add a speedometer
+        @speedometer = 0
+    end
+    # make a get_info method that prints a sentence with all the data of bike using string interpolation
+    def get_info
+        p "The #{@model} bike has #{@wheels} wheels and a #{@frame_size} cm frame."
+    end
+    # (3) create method that prints 'Beep! Beep!' when bell rings
+    def ring_bell
+        p @bell = 'Beep! Beep!'
+    end
+    # (4) get the current speedmeter reading
+    def speed
+        p @speedometer
+    end
+    # (5) create method to pedal faster that takes number as argument (which is the speed)
+    def pedal_faster speed
+        p @speedometer += speed
+    end
+    # (5) create method to brake BUT speed cannot be negative
+    def brake speed
+        @speedometer -= speed
+        # if speed is negative, set speed to zero regardless
+        if @speedometer < 0
+            p @speedometer = 0
+        else
+            p @speedometer
+        end
+    end
+end
+
+
+my_generic_bike = Bike.new('generic_bike', 'average_size')
+my_generic_bike.get_info
+my_cool_bike = Bike.new('Trek',163)
+my_cool_bike.get_info
 
```

Comment:
Looks really good, I would suggest printing the outputs on like 72, 74, etc rather than having the print inside the method itself. That was you know for sure you are getting the correct return.